src/components/AddPaymentCard/PaymentCardForm.tsx

```diff
@@ -129,58 +129,48 @@ function PaymentCardForm({
     headerContent,
 }: PaymentCardFormProps) {
     const styles = useThemeStyles();
+    const [data] = useOnyx(ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM);
+
     const {translate} = useLocalize();
     const route = useRoute();
     const label = CARD_LABELS[isDebitCard ? CARD_TYPES.DEBIT_CARD : CARD_TYPES.PAYMENT_CARD];
 
     const cardNumberRef = useRef<AnimatedTextInputRef>(null);
 
-    const [isCurrencyModalVisible, setIsCurrencyModalVisible] = useState(false);
-    const [currency, setCurrency] = useState<keyof typeof CONST.CURRENCY>(CONST.CURRENCY.USD);
-
-    const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM> => {
+    const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM> => {
         const errors = ValidationUtils.getFieldRequiredErrors(values, REQUIRED_FIELDS);
 
         if (values.nameOnCard && !ValidationUtils.isValidLegalName(values.nameOnCard)) {
-            errors.nameOnCard = translate('addDebitCardPage.error.invalidName');
+            errors.nameOnCard = label.error.nameOnCard;
         }
 
         if (values.cardNumber && !ValidationUtils.isValidDebitCard(values.cardNumber.replace(/ /g, ''))) {
-            errors.cardNumber = translate('addDebitCardPage.error.debitCardNumber');
+            errors.cardNumber = label.error.cardNumber;
         }
 
         if (values.expirationDate && !ValidationUtils.isValidExpirationDate(values.expirationDate)) {
-            errors.expirationDate = translate('addDebitCardPage.error.expirationDate');
+            errors.expirationDate = label.error.expirationDate;
         }
 
         if (values.securityCode && !ValidationUtils.isValidSecurityCode(values.securityCode)) {
-            errors.securityCode = translate('addDebitCardPage.error.securityCode');
+            errors.securityCode = label.error.securityCode;
         }
 
         if (values.addressStreet && !ValidationUtils.isValidAddress(values.addressStreet)) {
-            errors.addressStreet = translate('addDebitCardPage.error.addressStreet');
+            errors.addressStreet = label.error.addressStreet;
         }
 
         if (values.addressZipCode && !ValidationUtils.isValidZipCode(values.addressZipCode)) {
-            errors.addressZipCode = translate('addDebitCardPage.error.addressZipCode');
+            errors.addressZipCode = label.error.addressZipCode;
         }
 
         if (!values.acceptTerms) {
-            errors.acceptTerms = translate('common.error.acceptTerms');
+            errors.acceptTerms = 'common.error.acceptTerms';
```

Comment:
This change causes regression
![image](https://github.com/Expensify/App/assets/108292595/bd981152-3f4e-4fd9-8bec-399d660ee317)